WordPress/gutenberg
# [Navigation screen] Replace component state and refs with Redux state

## Description

This PR replaces the component state like `const [blocks, setBlocks] = useState();` and `const menuItemRef = useRef()` in favor of the redux state and data layer.

## How has this been tested?
1. Enable the experimental navigation screen
1. Go to `/wp-admin/admin.php?page=gutenberg-navigation` and interact with it in every way you can think of: create a new menu, add menu items, edit existing ones, remove some, save the menu, rinse and repeat without refreshing the page.
1. Confirm everything worked as expected.

## Types of changes
New feature

## Checklist:
